### PR TITLE
chore: Add CRD / api for nemo guardrail service

### PR DESCRIPTION
**Pull request by slu2011**
API / CRD / base controller code for guardrail service. 

Controller not hook up yet. Next PR would hook it up / add more tests. 

Remove the "storage" field as guard rail does not need model storage

Add "ConfigStorage" field for guard rail config (configmap or PVC)

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -62,6 +62,7 @@ rules:
 - apiGroups:
   - apps.nvidia.com
   resources:
+  - NemoGuardrailServices
```
Comment (shivamerla) [inline config/rbac/role.yaml:65]: this should just be `nemoguardrails`

File: api/apps/v1alpha1/nemo_guardrails_service_types.go [hunk 2]
```diff
@@ -0,0 +1,746 @@
...
+
+// NemoGuardrailServiceSpec defines the desired state of NemoGuardrailService
+type NemoGuardrailServiceSpec struct {
+	Image   Image           `json:"image,omitempty"`
+	Command []string        `json:"command,omitempty"`
+	Args    []string        `json:"args,omitempty"`
+	Env     []corev1.EnvVar `json:"env,omitempty"`
+	// The name of an existing pull secret containing the NGC_API_KEY
+	AuthSecret string `json:"authSecret"`
+	// Storage is the target storage for caching NIM model if NIMCache is not provided
```
Comment (shivamerla) [inline api/apps/v1alpha1/nemo_guardrails_service_types.go:62]: nit: comment have to be updated.

File: api/apps/v1alpha1/nemo_guardrails_service_types.go [hunk 3]
```diff
@@ -0,0 +1,746 @@
...
+)
+
+// NemoGuardrailServiceSpec defines the desired state of NemoGuardrailService
+type NemoGuardrailServiceSpec struct {
+	Image   Image           `json:"image,omitempty"`
+	Command []string        `json:"command,omitempty"`
+	Args    []string        `json:"args,omitempty"`
+	Env     []corev1.EnvVar `json:"env,omitempty"`
+	// The name of an existing pull secret containing the NGC_API_KEY
+	AuthSecret string `json:"authSecret"`
```
Comment (shivamerla) [inline api/apps/v1alpha1/nemo_guardrails_service_types.go:61]: is `AuthSecret` a required param for Guardrails? It is not pulling any models from NGC, but is this required for an API access from the API catalog endpoint?

File: api/apps/v1alpha1/nemo_guardrails_service_types.go [hunk 4]
```diff
@@ -0,0 +1,746 @@
...
+	}
+	params.HPASpec = hpaSpec
+	return params
+}
+
+// GetSCCParams return params to render SCC from templates
+func (n *NemoGuardrailService) GetSCCParams() *rendertypes.SCCParams {
+	params := &rendertypes.SCCParams{}
+	// Set metadata
+	params.Name = "nim-service-scc"
```
Comment (shivamerla) [inline api/apps/v1alpha1/nemo_guardrails_service_types.go:680]: nit: nim-guardrails-scc

File: api/apps/v1alpha1/nemo_guardrails_service_types.go (hunk 3 above)
Comment (slu2011) [reply api/apps/v1alpha1/nemo_guardrails_service_types.go]: Exactly, that is why I leave this field

**Review by shivamerla — APPROVED** [review]
LGTM! thanks @slu2011

**Pull request merged**